controller.go

```diff
@@ -225,6 +249,30 @@ func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver,
 	return nil
 }
 
+func (c *controller) RegisterIpam(name string, ipConfig ipamapi.Config, ipAllocator ipamapi.Allocator) error {
+	if !config.IsValidName(name) {
+		return ErrInvalidName(name)
+	}
+
+	c.Lock()
+	_, ok := c.ipams[name]
+	c.Unlock()
+	if ok {
+		return driverapi.ErrActiveRegistration(name)
+	}
+	l, g, err := ipConfig.GetDefaultAddressSpaces()
+	if err != nil {
+		return fmt.Errorf("ipam driver %s failed to return default address spaces: %v", name, err)
+	}
+	c.Lock()
+	c.ipams[name] = &ipamData{config: ipConfig, allocator: ipAllocator, defaultLocalAddressSpace: l, defaultGlobalAddressSpace: g}
```

Comment:
No, because libnetwork needs to know what are the ipam driver's default local/global address space strings. For the case where ipconfig (from UX) does not contain an explicitly chosen one which is the case for now.